src/index.d.ts

```diff
@@ -0,0 +1,693 @@
+import { Script, TxOut } from 'bsv';
+
+declare interface CastTemplate {
+  lockingScript: {
+    script: unknown[];
+    size: ((params: object) => number) | number;
+    validate: (params: object) => void;
+  }
+}
+
+/**
+ * Forge transaction builder class.
+ */
+declare class Forge {
+  /**
+   * Instantiates a new Forge instance.
+   * 
+   * The accepted params are:
+   * 
+   * * `inputs` - list of input objects or cast instances
+   * * `outputs` - list of output objects or cast instances
+   * * `changeTo` - address to send change to
+   * * `changeScript` - bsv Script object to send change to
+   * * `options` - set `rates` or `debug` options
+   * 
+   * @param params Tx parameters
+   * @constructor
+   */
+  constructor(params?: Partial<{
+    inputs: object[] | Cast[];
+    outputs: object[] | Cast[];
+    changeTo: string;
+    changeScript: unknown;
```

Comment:
The `changeScript` would be a Script instance from the bsv library. I don't think that library has type definitions yet - so maybe unknown is ok?

Reply:
I've typed it as `Script` just on the chance types are added to `bsv` in the future. It results in `any` which isn't ideal compared to `unknown` since `any` lacks the statement of intent, but both `unknown` and `any` work the same from the point of view of the developer using `txforge` so I wouldn't worry about it :relaxed: